perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,246 @@
+          end_date TIMESTAMP,
+          PRIMARY KEY (configuration_id) NOT ENFORCED
+        ) OPTIONS (description = 'Table for storing Job Configurations and respective VM instance name on which the job was run');
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID)
+
+    # API Request to create experiment_configuration if it does not exist
+    results = client.query(query_create_table_experiment_configuration)
+
+    query_check_end_date_update = """
+      SELECT configuration_id
+      FROM `{}.{}.{}`
+      WHERE gcsfuse_flags = '{}'
+      AND branch = '{}'
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID, gcsfuse_flags, branch)
+
+    results = client.query(query_check_end_date_update)
+
+    config_id = None
+    for row in results:
+      config_id = row['configuration_id']
+      query_update_end_date = """
+      UPDATE `{}.{}.{}`
+      SET end_date = '{}'
+      WHERE configuration_id = '{}'
+      """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID, end_date, config_id)
+      return config_id
+
+    query_get_configuration_id = """
+      SELECT configuration_id
+      FROM `{}.{}.{}`
+      WHERE gcsfuse_flags = '{}'
+      AND branch = '{}'
+      AND end_date = '{}'
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID, gcsfuse_flags, branch, end_date)
+
+    exists = False
+    query_job = client.query(query_get_configuration_id)
+    for row in query_job:
+      config_id = row['configuration_id']
+      exists = True
+    if not exists:
+      table_ref = dataset_ref.table(CONFIGURATION_TABLE_ID)
+      table = client.get_table(table_ref)
+      row_count = table.num_rows
+      config_id = row_count + 1
+      rows_to_insert = [(config_id, gcsfuse_flags, branch, end_date)]
+      results = client.insert_rows(table, rows_to_insert)
+
+    return int(config_id)
+
+  def _setup_bigquery(self):
+
+    """Creates the tables to store the metrics data if they don't exist in the dataset
+    """
+    # Query for creating fio_metrics table
+    query_create_table_fio_metrics = """
+          CREATE TABLE IF NOT EXISTS {}.{}.{}(
+            configuration_id INT64, 
+            start_time_build TIMESTAMP,
+            test_type STRING, 
+            num_threads INT64, 
+            file_size_kb INT64, 
+            block_size_kb INT64,
+            start_time INT64, 
+            end_time INT64, 
+            iops FLOAT64, 
+            bandwidth_bytes_per_sec INT64, 
+            IO_bytes INT64, 
+            min_latency FLOAT64, 
+            max_latency FLOAT64, 
+            mean_latency FLOAT64, 
+            percentile_latency_20 FLOAT64, 
+            percentile_latency_50 FLOAT64, 
+            percentile_latency_90 FLOAT64, 
+            percentile_latency_95 FLOAT64, 
+            FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+          ) OPTIONS (description = 'Table for storing FIO metrics extracted from periodic performance load testing');
```

Comment:
We should call the periodic performance load testing and experiments different. So, use experiments here.